docs/developer/index.rst

```diff
@@ -724,13 +769,135 @@ current way that models are checked:
+property attributes, so that they can apply the validation checks when
+the field is changed.  The intention is not to catch all possible
+problems, but to cover the obvious cases.
+
+.. _data_design_ndim:
+
+Data dimensionality
+^^^^^^^^^^^^^^^^^^^
+
+`~sherpa.data.Data` objects have a `~sherpa.data.Data.ndim` field,
+which is used to ensure that the model and data dimensions match when
+using the `~sherpa.data.Data.eval_model` and
+`~sherpa.data.Data.eval_model_to_fit` methods.
+
+.. _data_design_size:
+
+The size of a data object
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The `~sherpa.data.Data.size` field describes the size of a data
+object, that is the number of individual elements. Once a data object
+has its size set it can not be changed (this is new to Sherpa 4.14.1,
+as in previous versions you could change fields to any size). This
+field can also be accessed using `len`, with it returning 0 when no
+data has been set.
+
+.. _data_design_point_vs_integrated:
+
+Point versus Integrated
+^^^^^^^^^^^^^^^^^^^^^^^
+
+There is currently no easy way to identify whether a data object
+requires integrated (low and high edges) or point axes (the coordinate
+at which to evaluate the model).
+
+.. _data_design_independent_axis:
+
+Handling the independent axis
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Checks have been added in Sherpa 4.14.1 to ensure that the correct
+number of arrays are used when setting the independent axis: that is,
+a `~sherpa.data.Data1D` object uses `(x,)`, `~sherpa.data.Data1DInt`
+uses `(lo, hi)`, and `~sherpa.data.Data2D` uses `(x0, x1)`. Note that
+the arguement is expected to be a tuple, even in the
+`~sherpa.data.Data1D` case, and that the individual components are
+checked to ensure they have the same size.
+
+The handling of the independent axis is mediated by a "Data Space"
+object (`~sherpa.data.DataSpaceND`, `~sherpa.data.DataSpace1D`,
+`~sherpa.data.IntegratedDataSpace1D`, `~sherpa.data.DataSpace2D`, and
+`~sherpa.data.IntegratedDataSpace2D`) which is handled by the
+`_init_data_space` and `_check_data_space` methods of the
+`~sherpa.data.Data` class.
+
+To ensure that any filter remains valid, the independent axis is
+marked as read-only. The only way to change a value is to change the
+whole independent axis, in which case the code recognizes that the
+filter - whether just the `~sherpa.data.Data.mask` attribute or also
+any region filter for the `~sherpa.astro.data.DataIMG` case - has to
+be cleared.
+
+.. _data_design_validation:
+
+Validation
+^^^^^^^^^^
+
+Fields are converted to `ndarray` - if not `None` - and then checked
+to see if they are 1D and have the correct size. Some fields may have
+extra checks, such as the `~sherpa.astro.data.DataPHA.grouping` and
+`~sherpa.astro.data.DataPHA.quality` columns for PHA data which
+are converted to integer values.
+
+There is limited support for NumPy masked arrays, but the assumption
+is that and data has been masked before being added to a
+`~sherpa.data.Data` object.
+
```

Comment:
The sentences in borked (technical term in British English) and it's also explained elsewhere, so just delete?

```suggestion

```